web/src/utils/usePaginatorApi.ts

```diff
@@ -55,6 +55,13 @@ export const usePaginatorApi = <T extends (...args: any[]) => any>({
 
   const [pageOptions, setPageOptions] = useState<number[]>([pageIndex]);
 
+  useEffect(() => {
+    if (data && data.next && pageOptions.length == 1) {
```

Comment:
```suggestion
    if (data?.next && pageOptions?.length === 1) {
```
Not sure but I think we can use [optional changing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html). I think our typescript version supports this. 

Not sure if `pageOptions` returns an empty array by default; if not then it will cause an error when trying to get `length` of `undefined` 

`===` instead of `==`